### PR TITLE
[MM-29666] compliance export skipped posts

### PR DESCRIPTION
**Pull request by maxerenberg**
#### Summary
This PR fixes a bug in the compliance export which causes some posts to be skipped when they have the same timestamp.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-29666

#### Release Note
```release-note
NONE
```

#### Enterprise PR
https://github.com/mattermost/enterprise/pull/977

### CODE REVIEW

**Comment by maxerenberg** [comment]
This cause of the bug was the exact same problem we were having in https://github.com/mattermost/mattermost-server/pull/17185 - we assumed that timestamps are unique for each post, which they are not.

**Review by sbishel — CHANGES_REQUESTED** [review]

File: jobs/jobs_watcher.go [hunk 1]
```diff
@@ -84,6 +84,7 @@ func (watcher *Watcher) PollAndNotify() {
 				select {
 				case watcher.workers.MessageExport.JobChannel() <- *job:
 				default:
+					panic("Couldn't send job to worker")
```
[the remark this answers is not in the document]
Comment (maxerenberg) [reply jobs/jobs_watcher.go:87]: Whoops! Yes, this was left over from debugging.

**Review by sbishel — APPROVED** [review]

**Review by mkraft — APPROVED** [review]

**Comment by agnivade** [comment]
Great work with this! I think you have solved our Message reliability problem with this query :laughing: 

Although, we need to be careful with this one. This is effectively the same as https://github.com/mattermost/mattermost-server/pull/16527, which does a two-column ordering, and due to that the query performance dropped drastically (https://github.com/mattermost/mattermost-server/pull/16527#discussion_r553928532).

And that is precisely why that PR got paused. Message export performance is important to a lot of customers.

Ideally, I'd like to see the query fixed, and the performance remain the same. That needs cleaning up redundant indexes, and adding new ones I believe. Which means writing asynchronous one-off scripts to add/remove indexes that the customers can run before upgrading Mattermost. But if this needs to go in urgently, that's okay too. Just that we all need to be aware of the tradeoffs.

**Review by furqanmlk — COMMENTED** [review]
**Following scenarios are failing** 
Scenario 01:
1. Create 50 posts with slash command
2. Create 5 posts using API with same timestamp
3. Run Compliance Export
4. It exports only 50 posts that were posted using slash command, it does not export posts from API with same time stamp
Note: API posts are in Datebase and are shown in Channels.

Scenario 02 : Any post by API is not being exported

**Pull request merged**